Libplanet.RocksDBStore/RocksDBStore.cs

```diff
@@ -338,15 +378,27 @@ public override Transaction<T> GetTransaction<T>(TxId txid)
             }
 
             byte[] key = TxKey(txid);
-            byte[] bytes = _txDb.Get(key);
+            byte[] epochBytes = _txIndexDb.Get(key);
 
-            if (bytes is null)
+            if (epochBytes is null)
             {
                 return null;
             }
 
-            Transaction<T> tx = Transaction<T>.Deserialize(bytes, false);
+            RocksDBStoreEpoch epoch = RocksDBStoreEpoch.Deserialize(epochBytes);
+
+            _rwTxLock.EnterReadLock();
+            if (!_txDbCache.TryGetValue(epoch.DbName, out RocksDb txDb))
+            {
+                txDb = RocksDBUtils.OpenRocksDb(_options, TxDbPath(epoch.DbName));
+                _txDbCache.AddOrUpdate(epoch.DbName, txDb);
+            }
+
+            byte[] txBytes = txDb.Get(key);
+
+            Transaction<T> tx = Transaction<T>.Deserialize(txBytes, false);
             _txCache.AddOrUpdate(txid, tx);
+            _rwTxLock.ExitReadLock();
```

Comment:
`_rwTxLock.ExitReadLock()` should be inside a `finally` block so that it's guaranteed to be called even if an any error happens.

```suggestion
            try
            {
                if (!_txDbCache.TryGetValue(epoch.DbName, out RocksDb txDb))
                {
                    txDb = RocksDBUtils.OpenRocksDb(_options, TxDbPath(epoch.DbName));
                    _txDbCache.AddOrUpdate(epoch.DbName, txDb);
                }

                byte[] txBytes = txDb.Get(key);

                Transaction<T> tx = Transaction<T>.Deserialize(txBytes, false);
                _txCache.AddOrUpdate(txid, tx);
            }
            finally
            {
                _rwTxLock.ExitReadLock();
            }
```